docs/internal/singleregion.md

```diff
@@ -0,0 +1,257 @@
+# Single active region types
+
+Verona has a concept of regions as its core way of reasoning about ownership.
+There are multiple systems that could be developed to enable this to work soundly.
+In this document, we develop having a single active region that may be mutated at any point in time.
+
+This document is a working document and should not be consider definitive.  It represents the best current view of how the type system should work.
+
+## Informal
+
+Verona is a structurally type and algebraic programming language.  It is an object based language, where the only values are references to objects.  There are some primitive types that the runtime represents more efficiently, but abstractly everything can be viewed as an object.
+
+Verona's core concept for ownership is regions.  A region is a group of objects.
+There are two types of region in Verona, mutable and immutable.  There is a single immutable region.  The immutable region is closed under flowing fields from objects in that region.
+```
+  ∀ x,f. region_of(x) = immutable ⇒ region_of(x.f) = immutable
+```
+There can be multiple mutable regions.  There is a single object in a mutable region that is the entry point. There is a single reference to the entry point from outside the region.  There may be multiple references from within the region to any other object, including the entry point.  This can be enforced by:
+```
+  ∀ ref1,ref2.
+    ref1.src ≠ ref2.src ∧
+    region_of(ref1.dst) = region_of(ref2.dst) ∧
+    region_of(ref1.dst) ≠ immutable
+    ⇒
+      region_of(ref1.src) = region_of(ref1.dst) ∨
+      region_of(ref2.src) = region_of(ref1.dst)

```

Comment:
@sophiaIC is this better for you?

Reply:
actually no, what I wanted is

      region_of(ref1.src) = region_of(ref1.dst) ∨
      region_of(ref2.src) = region_of(**ref2**.dst)
      
which says, that `ref1` or `ref2` do not ``cross regions''

Reply:
sorry, it is showing an old diff. I have changed it like that.  Hopefully added a new comment to the correct revision